# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraphImpl.h
# Commit: 96f574da3bbb
# Full Hash: 96f574da3bbbaa7c7d5f5000f1f99fd3efb11967
# Author: Paul Adenot <paul@paul.cx>
# Date: 2020-01-24 03:50:24
# Regressor Bug: 1608505
# File Overlap Count: 1
# Description:
#   Bug 1608505 - Cap the audio output channel count to something the device can handle. r=achronop
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D60549
# ==============================================================================

diff -r 9fb3bfa5ea09 -r 96f574da3bbb dom/media/MediaTrackGraphImpl.h
--- a/dom/media/MediaTrackGraphImpl.h	Thu Jan 23 15:51:22 2020 +0000
+++ b/dom/media/MediaTrackGraphImpl.h	Thu Jan 23 15:18:43 2020 +0000
@@ -485,7 +485,10 @@
     mTrackOrderDirty = true;
   }
 
+  // Get the current maximum channel count. Graph thread only.
   uint32_t AudioOutputChannelCount() const;
+  // Set a new maximum channel count. Graph thread only.
+  void SetMaxOutputChannelCount(uint32_t aMaxChannelCount);
 
   double AudioOutputLatency();
 
@@ -997,6 +1000,14 @@
    * whenever the audio device running this MediaTrackGraph changes.
    */
   double mAudioOutputLatency;
+
+  /**
+   * The max audio output channel count the default audio output device
+   * supports. This is cached here because it can be expensive to query. The
+   * cache is invalidated when the device is changed. This is initialized in the
+   * ctor, and the read/write only on the graph thread.
+   */
+  uint32_t mMaxOutputChannelCount;
 };
 
 }  // namespace mozilla
